src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -1315,6 +1321,19 @@ else if ((sym.kind == TYP ||
         return flags & (mask | ~ExtendedStandardFlags) | implicit;
     }
 
+    private void warnOnExplicitStrictfp(DiagnosticPosition pos, JCTree tree) {
+        DiagnosticPosition prevLintPos = deferredLintHandler.setPos(tree.pos());
+        try {
```

Comment:
Do we need tree passed as a parameter at all ? Why can't we use just pos instead of tree.pos() Looking at the 3 call sites of checkFlags it seems guaranteed that pos is same as tree.pos() ?

Reply:
Yes; tree argument is unnecessary, removed in a subsequent push.